geo-types/src/private_utils.rs

```diff
@@ -72,14 +72,15 @@ where
     }
     let dx = end.x - start.x;
     let dy = end.y - start.y;
-    let r = ((point.x - start.x) * dx + (point.y - start.y) * dy) / (dx.powi(2) + dy.powi(2));
+    let d = dx * dx + dy * dy;
+    let r = ((point.x - start.x) * dx + (point.y - start.y) * dy) / d;
     if r <= T::zero() {
         return line_euclidean_length(Line::new(point, start));
     }
     if r >= T::one() {
         return line_euclidean_length(Line::new(point, end));
     }
-    let s = ((start.y - point.y) * dx - (start.x - point.x) * dy) / (dx * dx + dy * dy);
+    let s = ((start.y - point.y) * dx - (start.x - point.x) * dy) / d;
     s.abs() * dx.hypot(dy)
```

Comment:
```suggestion
    s.abs() * d.sqrt()
```

right? I have no idea if it's faster though.

Reply:
It might be faster, but it's probably less precise. Compare that to the [previous and current](https://github.com/bminor/glibc/commit/6c848d70383e1dbe932ef41723ac0abfdeec7ca8) glibc implementations.

Reply:
Now that you say it, I'm not surprised, but wow, there are so many gotchas with this stuff!